operator/pkg/service/annotations.go

```diff
@@ -36,3 +36,39 @@ func GetLBProtocolAnnotation(obj metav1.Object) string {
 func GetLBProtocolModelAnnotation(obj metav1.Object) string {
 	return obj.GetAnnotations()[lbModeAnnotation]
 }
+
+//
+// ClusterMutator functions
+//
+
+//
+// HTTPConnectionManagerMutator functions
+//
+
+// grpcHttpConnectionManagerMutator returns a function that mutates the upgradeConfigs for grpc protocol
+func grpcHttpConnectionManagerMutator(obj metav1.Object) httpConnectionManagerMutator {
+	return func(manager *http_connection_manager_v3.HttpConnectionManager) *http_connection_manager_v3.HttpConnectionManager {
+		if GetLBProtocolAnnotation(obj) == "grpc" {
```

Comment:
Is it possible to automatically configure the `UpgradeConfig` based on the incoming request instead of requiring it to be explicitly enabled?

Reply:
oh right, I remembered that you mentioned this point before, totally forgot about it. Let me check and get back to you later.